lib/sshkit/context.ex

```diff
@@ -1,10 +1,24 @@
 defmodule SSHKit.Context do
-  @moduledoc false
+  @moduledoc """
+  Holds information about the context in which a command can be executed.
+  This includeds the `group`, `user`, `umask` setting, `path`,
+  environment variables, as well as the hosts a command should be on.
+
+  `build/2` compiles a context into an executable command string.
+  """
 
   import SSHKit.Utils
 
   defstruct [hosts: [], env: nil, path: nil, umask: nil, user: nil, group: nil]
 
+  @doc """
+  Compile the given context into a string that is executable (via SSH) on a shell.
+
+  ## Parameters
+
+  * `context`: a `SSHKit.Context` struct
+  * `command`: a string containing the command to execute in that context
+  """
```

Comment:
Also, how about:

```
Compiles an executable command string for running the given `command` in the provided `context`.
```

This seems more like how the official Elixir libs are documented. Instead of stating the parameter types in the docstring, we could rather provide an example there:

```
## Example

` ``
iex> %SSHKit.Context{path: "/var/www"} |> SSHKit.Context.build("ls")
"cd /var/www && /usr/bin/env ls"
` ``
```

and add proper typespecs (https://github.com/bitcrowd/sshkit.ex/issues/49).

– see [`Path`](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/path.ex) docs for an example